packages/dapp-svelte-wallet/api/src/wallet.js

```diff
@@ -49,167 +50,150 @@ export function buildRootObject(_vatPowers) {
           },
-          error(offer, reason) {
-            httpSend(
-              {
-                type: 'walletOfferResult',
-                data: {
-                  id: offer.id,
-                  dappContext: offer.dappContext,
-                  error: `${(reason && reason.stack) || reason}`,
-                },
-              },
-              [meta.channelHandle],
-            );
+          finish(offers) {
+            updater.finish(offers.filter(filter));
           },
-          handled(offer) {
-            if (handled) {
-              return;
-            }
-            handled = true;
-            httpSend(
-              {
-                type: 'walletOfferHandled',
-                data: offer.id,
-              },
-              [meta.channelHandle],
-            );
+          fail(e) {
+            updater.fail(e);
           },
         });
-        return {
-          type: 'walletOfferAdded',
-          data: await wallet.addOffer(
-            { ...data, actions },
-            { ...meta, dappOrigin },
-          ),
-        };
-      }
-      case 'walletDeclineOffer': {
-        return {
-          type: 'walletDeclineOfferResponse',
-          data: await wallet.declineOffer(data),
-        };
-      }
-      case 'walletCancelOffer': {
-        return {
-          type: 'walletCancelOfferResponse',
-          data: await wallet.cancelOffer(data),
-        };
-      }
-      case 'walletAcceptOffer': {
-        const { outcome } = await wallet.acceptOffer(data);
-        // We return the outcome only if it is a string. Otherwise we
-        // return a default message.
-        const result =
-          typeof outcome === 'string'
-            ? { outcome }
-            : { outcome: 'Offer was made.' };
-        return {
-          type: 'walletAcceptOfferResponse',
-          data: result,
-        };
-      }
+        return notifier;
+      },
+      async getDepositFacetId(brandBoardId) {
+        await approve();
+        return walletInternals.getDepositFacetId(brandBoardId);
+      },
+      async suggestIssuer(petname, boardId) {
+        await approve();
+        return walletInternals.suggestIssuer(petname, boardId, dappOrigin);
+      },
+      async suggestInstallation(petname, boardId) {
+        await approve();
+        return walletInternals.suggestInstallation(
+          petname,
+          boardId,
+          dappOrigin,
+        );
+      },
+      async suggestInstance(petname, boardId) {
+        await approve();
+        return walletInternals.suggestInstance(petname, boardId, dappOrigin);
+      },
+    };
+    return harden(bridge);
+  };
 
-      case 'walletGetOffers':
-      case 'walletGetOfferDescriptions': {
-        const result = await wallet.getOffers({ origin: dappOrigin });
-        return {
-          type: 'walletOfferDescriptions',
-          data: result,
-        };
-      }
+  /** @type {WalletBridge} */
+  const anonymousBridge = {
```

Comment:
Can we add a comment about when an anonymousBridge should be used and what makes it different?

Reply:
Looks like it is when there is no `dappOrigin`?

Reply:
I'll rename it `powerfulBridge`.  It's the bridge that is accessible directly to the REPL and deploy scripts (i.e. things that bypass the per-dapp-origin access control).

Reply:
It's now `preapprovedBridge`.